src/main/java/org/folio/event/handler/HoldingCreateAsyncRecordHandler.java

```diff
@@ -0,0 +1,118 @@
+package org.folio.event.handler;
+
+import static org.folio.event.InventoryEventType.INVENTORY_HOLDING_CREATE;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.folio.event.dto.InventoryFields;
+import org.folio.event.dto.ResourceEvent;
+import org.folio.event.service.AuditOutboxService;
+import org.folio.models.ConsortiumConfiguration;
+import org.folio.okapi.common.GenericCompositeFuture;
+import org.folio.rest.jaxrs.model.Location;
+import org.folio.rest.jaxrs.model.OrderLineAuditEvent;
+import org.folio.rest.jaxrs.model.Piece;
+import org.folio.rest.jaxrs.model.PieceAuditEvent;
+import org.folio.rest.jaxrs.model.PoLine;
+import org.folio.rest.persist.Conn;
+import org.folio.rest.persist.DBClient;
+import org.folio.services.consortium.ConsortiumConfigurationService;
+import org.folio.services.lines.PoLinesService;
+import org.folio.services.piece.PieceService;
+import org.folio.spring.SpringContextUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class HoldingCreateAsyncRecordHandler extends InventoryCreateAsyncRecordHandler {
+
+  @Autowired
+  private PieceService pieceService;
+
+  @Autowired
+  private PoLinesService poLinesService;
+
+  @Autowired
+  private ConsortiumConfigurationService consortiumConfigurationService;
+
+  @Autowired
+  private AuditOutboxService auditOutboxService;
+
+  public HoldingCreateAsyncRecordHandler(Vertx vertx, Context context) {
+    super(INVENTORY_HOLDING_CREATE, vertx, context);
+    SpringContextUtil.autowireDependencies(this, context);
+  }
+
+  @Override
+  protected Future<Void> processInventoryCreationEvent(ResourceEvent resourceEvent, String tenantId, Map<String, String> headers) {
+    var holdingObject = JsonObject.mapFrom(resourceEvent.getNewValue());
+    var holdingId = holdingObject.getString(InventoryFields.ID.getValue());
+    return new DBClient(getVertx(), tenantId).getPgClient()
+      .withTrans(conn -> {
+        var tenantIdUpdates = List.of(
+          processPoLinesUpdate(holdingId, tenantId, headers, conn),
+          processPiecesUpdate(holdingId, tenantId, headers, conn)
+        );
+        return GenericCompositeFuture.all(tenantIdUpdates).mapEmpty();
+      });
```

Comment:
we need to add .onComplete section to process transaction outbox tables, see usages of "auditOutboxService.processOutboxEventLogs(okapiHeaders);"